Week3/homework/step2-3.js

```diff
@@ -18,8 +19,9 @@ function repeatStringNumTimesWithWhile(str, num) {
   // eslint-disable-next-line
   let result = '';
 
-  // Replace this comment and the next line with your code
-  console.log(str, num, result);
+  while (result.length < str.length * num) {
```

Comment:
The loop condition requires computation (property access, multiplication) for each iteration. It would be more efficient to introduce a loop index variable and simplify the loop condition, e.g.:

```js
let i = 0;
while (i < num) {
  //...
  i++;
}
```

This mimics how the `for` loop works.